public/app/features/transformers/timeSeriesTable/timeSeriesTableTransformer.ts

```diff
@@ -0,0 +1,125 @@
+import { map } from 'rxjs/operators';
+
+import {
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  DataTransformerInfo,
+  Field,
+  FieldType,
+  Labels,
+  MutableDataFrame,
+} from '@grafana/data';
+
+export interface TimeSeriesTableTransformerOptions {}
+
+export const timeSeriesTableTransformer: DataTransformerInfo<TimeSeriesTableTransformerOptions> = {
+  id: DataTransformerID.timeSeriesTable,
+  name: 'Time series to table transform',
+  description: 'Time series to table rows',
+  defaultOptions: {},
+
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        return timeSeriesToTableTransform(options, data);
+      })
+    ),
+};
+
+export function timeSeriesToTableTransform(options: TimeSeriesTableTransformerOptions, data: DataFrame[]): DataFrame[] {
+  const refId2LabelFields = getLabelFields(data);
+
+  const refId2frameField: Record<string, Field<DataFrame, ArrayVector>> = {};
+
+  const result: DataFrame[] = [];
+
+  for (const frame of data) {
+    if (!isTimeSeries(frame)) {
+      result.push(frame);
+    }
+
+    const refId = frame.refId ?? '';
+
+    const labelFields = refId2LabelFields[refId] ?? {};
+    let frameField = refId2frameField[refId];
+    if (!frameField) {
+      frameField = {
+        name: 'Trend' + (refId && Object.keys(refId2LabelFields).length > 1 ? ` #${refId}` : ''),
+        type: FieldType.frame,
+        config: {},
+        values: new ArrayVector(),
+      };
+      refId2frameField[refId] = frameField;
+      const table = new MutableDataFrame();
+      for (const label of Object.values(labelFields)) {
+        table.addField(label);
+      }
+      table.addField(frameField);
+      table.refId = refId;
+      result.push(table);
```

Comment:
not sure what this refId logic is doing, can you add a test that helps describe the refId logic? There can be multiple data frames with same refId

Reply:
- [ ] add tests, comments to transform